src/protocols/user_directed_commissioning/UserDirectedCommissioningServer.cpp

```diff
@@ -66,7 +66,54 @@ void UserDirectedCommissioningServer::OnMessageReceived(const Transport::PeerAdd
     ChipLogProgress(AppServer, "UDC instance=%s ", id.GetInstanceName());
 
     UDCClientState * client = mUdcClients.FindUDCClientState(instanceName);
-    if (client == nullptr)
+    if (id.GetCancelPasscode())
+    {
+        ChipLogProgress(AppServer, "UDC instance cancel received");
+        if (client == nullptr)
```

Comment:
ah, I see it is used more below. This is hard to actually follow. Can we break out this method that is over 130 lines long and seems to do competely distinct things (as each if has its own return)? Otherwise this is hard to read and maintain.

Reply:
addressed